conans/client/conan_api.py

```diff
@@ -78,6 +78,11 @@ def wrapper(api, *args, **kwargs):
             api.create_app(quiet_output=quiet_output)
             log_command(f.__name__, kwargs)
             with environment_append(api.app.cache.config.env_vars):
+                if (f.__name__ != "get_remote_by_name"):
```

Comment:
We have been discussing and this doesn't seem the best place for this. We'd rather have it in:

```python
    def create_app(self, quiet_output=None):
        self.app = ConanApp(self.cache_folder, self.user_io, self.http_requester,
                            self.runner, quiet_output=quiet_output)
        self.app.hook_manager.execute("init")
```

Reply:
It'd call the `init` twice for the `conan search` command in the current design then since each call to the api creates a separate app. 

So we must do an anti "rattle" mechanism for the hook_manager? That could be quite unexpected in some cases too. It's related to the previous comment:

> The main reason for this hook is to be able to do a `conan config install` with a given frequency 

How to expose frequency customization? What if a shop wants to do something on each call? 

We can implement a check to avoid recursion, that's not a problem. But I believe the frequency should be handled by hooks themselves. We probably need an example for that though.

Reply:
Frequency should be handled by hooks, internally, indeed, not by Conan.